### PR TITLE
Type schema

### PR DESCRIPTION
**Pull request by mickaelistria**
Add metadata to improve usability of extension points in PDE editor.

Signed-off-by: Mickael Istria <mistria@redhat.com>

### CODE REVIEW

File: org.eclipse.tm4e.registry/schema/grammars.exsd [hunk 1]
```diff
@@ -1,112 +1,118 @@
...
+      </annotation>
+      <complexType>
+         <attribute name="scopeName" type="string" use="required">
+            <annotation>
+               <documentation>
+                  TextMate scope name.
+               </documentation>
+            </annotation>
+         </attribute>
+         <attribute name="path" type="string">
```
Comment (angelozerr) [inline org.eclipse.tm4e.registry/schema/grammars.exsd]: Please remove this attribute. It was an error with copy/paste.

File: org.eclipse.tm4e.registry/schema/grammars.exsd [hunk 2]
```diff
@@ -1,112 +1,118 @@
...
-      <documentation>
-         This plugin itself does not have any predefined builders.
-      </documentation>
-   </annotation>
-
-
-</schema>
+<?xml version='1.0' encoding='UTF-8'?>
+<!-- Schema file written by PDE -->
+<schema targetNamespace="org.eclipse.tm4e.core" xmlns="http://www.w3.org/2001/XMLSchema">
```
Comment (angelozerr) [inline org.eclipse.tm4e.registry/schema/grammars.exsd]: targetNamespace should not be "org.eclipse.tm4e.registry" now?

File: org.eclipse.tm4e.registry/schema/grammars.exsd [hunk 3]
```diff
@@ -1,112 +1,118 @@
...
+<annotation>
+      <appinfo>
+         <meta.schema plugin="org.eclipse.tm4e.core" id="grammars" name="TextMate grammars"/>
```
Comment (angelozerr) [inline org.eclipse.tm4e.registry/schema/grammars.exsd]: plugin should not be "org.eclipse.tm4e.registry" now?

File: org.eclipse.tm4e.registry/schema/grammars.exsd (hunk 1 above)
Comment (mickaelistria) [inline org.eclipse.tm4e.registry/schema/grammars.exsd]: I'm not sure I get it. What's the error here?
PS: to see changes ignoring whitespace, you can append `?w=1` to the URL, such as https://github.com/eclipse/tm4e/pull/84/files?w=1

File: org.eclipse.tm4e.registry/schema/grammars.exsd (hunk 3 above)
Comment (mickaelistria) [inline org.eclipse.tm4e.registry/schema/grammars.exsd]: Yes, probably. However, can we track it in a separate change?

File: org.eclipse.tm4e.registry/schema/grammars.exsd (hunk 1 above)
Comment (angelozerr) [inline org.eclipse.tm4e.registry/schema/grammars.exsd]: It's not your error. It's my fault. "path" attribute has wrong label. I will fix it.

**Comment by angelozerr** [comment]
Thanks @mickaelistria !

**Pull request merged**